lib/spack/spack/spec.py

```diff
@@ -3063,7 +3137,6 @@ def virtual_dependencies(self):
         return [spec for spec in self.traverse() if spec.virtual]
 
     @property
-    @lang.memoized
```

Comment:
I suspect the failures is due to this deletion. Rebasing over the tip of `develop` I had a failure in the `clingo` solver tests related to this memoization (infinite recursion when trying a round trip to YAML for `py-flake8`). I currently:
1. Don't fully understand why the tests failed in the first place 
2. Nor I see why other tests are failing now without memoization in the original solver

Need some :sleeping: will be on it tomorrow.

Reply:
The bug in the unit tests is also present on the `develop` branch, it's transient and Python 3.5 specific (thus very likely to be related to dict iteration order). We see it here because it's exposed by a unit test added in this branch.